### PR TITLE
ZCS-3093:error during tgz import results in endless loop and memory leak

### PR DESCRIPTION
**Pull request by sneha-patil-synacor**
Issue:
Truncated contacts tgz import results in endless loop and memory leak

Fix:
Handling EOFException thrown by  ArchiveFormatter.readArchiveEntry() which will throw the ServiceException and the loop will not continue.

Testing done:
Manual testing. Added Junit test.

### CODE REVIEW

File: store/src/java-test/com/zimbra/cs/service/ModifyPrefsTest.java [hunk 1]
```diff
@@ -36,12 +36,10 @@
 import com.zimbra.common.util.L10nUtil;
 import com.zimbra.cs.account.Account;
 import com.zimbra.cs.account.Provisioning;
-import com.zimbra.cs.mailbox.MailItem;
```
Comment (shrikant-prasad) [inline store/src/java-test/com/zimbra/cs/service/ModifyPrefsTest.java]: these changes have already been done in ZCS2948.

Comment (sneha-patil-synacor) [reply store/src/java-test/com/zimbra/cs/service/ModifyPrefsTest.java]: Reverted

File: store/src/java-test/com/zimbra/cs/mailbox/ContactTest.java [hunk 2]
```diff
@@ -293,4 +303,22 @@ public void testEncodeContact() throws Exception {
         ToXML.encodeContact(response, new ItemIdFormatter(), new OperationContext(acct), contact, true, null);
         Assert.assertEquals(response.getElement("cn").getElement("a").getText(), "Cert1149638887753217");
     }
+
+    @Test
+    public void testTruncatedContactsTgzImport() throws IOException {
+        File file = new File("src/java-test/Truncated.tgz");
```
Comment (ykomiyam) [inline store/src/java-test/com/zimbra/cs/mailbox/ContactTest.java]: Truncated.tgz is missing; need to commit the file as well.

File: store/src/java/com/zimbra/cs/service/formatter/ArchiveFormatter.java [hunk 3]
```diff
@@ -957,6 +958,8 @@ public void saveCallback(UserServletContext context, String contentType, Folder
                         }
                         try {
                             id = new ItemData(readArchiveEntry(ais, aie));
+                        } catch (EOFException e) {
+                            throw ServiceException.FAILURE("Error reading file", e);
```
Comment (ykomiyam) [inline store/src/java/com/zimbra/cs/service/formatter/ArchiveFormatter.java]: Is there any particular reason to catch not the IOException but only the EOFException?  The "readArchiveEntry" may throw some other subclass of IOException which would also cause some endless loop.

File: store/src/java-test/com/zimbra/cs/mailbox/ContactTest.java (hunk 2 above)
Comment (sneha-patil-synacor) [reply store/src/java-test/com/zimbra/cs/mailbox/ContactTest.java]: Done

File: store/src/java/com/zimbra/cs/service/formatter/ArchiveFormatter.java (hunk 3 above)
Comment (sneha-patil-synacor) [reply store/src/java/com/zimbra/cs/service/formatter/ArchiveFormatter.java]: Done

**Pull request closed without merging**